Description: reduce the number of exported symbols in libQt5WebKit.so.5
Author: Dmitry Shachnev <mitya57@debian.org>
Forwarded: not-needed
Last-Update: 2017-08-27

--- a/Source/WTF/wtf/ExportMacros.h
+++ b/Source/WTF/wtf/ExportMacros.h
@@ -56,7 +56,7 @@
 
 #elif defined(__GNUC__) && !defined(__CC_ARM) && !defined(__ARMCC__)
 
-#define WTF_EXPORT_DECLARATION __attribute__((visibility("default")))
+#define WTF_EXPORT_DECLARATION
 #define WTF_IMPORT_DECLARATION WTF_EXPORT_DECLARATION
 #define WTF_HIDDEN_DECLARATION __attribute__((visibility("hidden")))
 
--- a/Source/cmake/OptionsQt.cmake
+++ b/Source/cmake/OptionsQt.cmake
@@ -658,6 +658,7 @@ if (COMPILER_IS_GCC_OR_CLANG AND UNIX)
     if (NOT SHARED_CORE)
         set(CMAKE_C_FLAGS "-fvisibility=hidden ${CMAKE_C_FLAGS}")
         set(CMAKE_CXX_FLAGS "-fvisibility=hidden -fvisibility-inlines-hidden ${CMAKE_CXX_FLAGS}")
+        add_definitions(-DJS_NO_EXPORT)
     endif ()
 endif ()
 

